synapse/storage/data_stores/main/stream.py

```diff
@@ -253,6 +257,18 @@ class StreamWorkerStore(EventsWorkerStore, SQLBaseStore):
     def __init__(self, database: Database, db_conn, hs):
         super(StreamWorkerStore, self).__init__(database, db_conn, hs)
 
+        self._instance_name = hs.get_instance_name()
+        self._send_federation = hs.should_send_federation()
+        self._federation_shard_config = hs.config.federation.federation_shard_config
+
+        cur = LoggingTransaction(
+            db_conn.cursor(),
+            name="_reset_federation_positions_txn",
+            database_engine=self.database_engine,
+        )
+        self._reset_federation_positions_txn(cur)
+        cur.close()
```

Comment:
Should this be wrapped in a `try-finally`? Or I guess if this fails the process goes down anyway so, whatever.

Reply:
Yeah, basically that. We don't do it anywhere else so :man_shrugging:

Reply:
Alright. That sounds fine then!